File: 2-live-code/5-border-box/solution.js

function generateBorderBox (N) {
  for (let i = 1; i < N + 1; i++) {
    let line = ``;
    for (let j = 1; j < N + 1; j++) {
      if (i == 1 || i == N || j == 1 || j == N) {
        line += `I`;
      } else if (
        (i == 3 || i == N-2 || j == 3 || j == N-2) &&
        (i >= 3 && i <= N-2) &&
        (j >= 3 && j <= N-2)) {
        line += `I`;
      } else if (i == Math.ceil(N/2) && j == Math.ceil(N/2)) {
        line += `*`;
      } else {
        line += ` `;
      }
    }
    console.log(line);
  }
}

generateBorderBox(9);
/*
IIIIIIIII
I       I
I IIIII I
I I   I I
I I * I I
I I   I I
I IIIII I
I       I
IIIIIIIII
*/

generateBorderBox(13);
/*
IIIIIIIIIIIII
I           I
I IIIIIIIII I
I I       I I
I I       I I
I I       I I
I I   *   I I
I I       I I
I I       I I
I I       I I
I IIIIIIIII I
I           I
IIIIIIIIIIIII
*/

generateBorderBox(23);
/*
IIIIIIIIIIIIIIIIIIIIIII
I                     I
I IIIIIIIIIIIIIIIIIII I
I I                 I I
I I                 I I
I I                 I I
I I                 I I
I I                 I I
I I                 I I
I I                 I I
I I                 I I
I I        *        I I
I I                 I I
I I                 I I
I I                 I I
I I                 I I
I I                 I I
I I                 I I
I I                 I I
I I                 I I
I IIIIIIIIIIIIIIIIIII I
I                     I
IIIIIIIIIIIIIIIIIIIIIII
*/

File: 2-live-code/5-border-box/solution.py
def generateBorderBox(N):
  mid = N // 2 + 1
  print(mid)
  for i in range(1, N + 1):
    line = ''
    for j in range(1, N + 1):
      # star in middle
      if i == mid and j == mid:
        line += '*'
      elif (i == 1) or (i == N) or (j == 1) or (j == N):
        line += 'I'
      elif ((i == 3) or (i == N - 2)) and j != 2 and j != N-1:
        line += 'I'
      elif ((j == 3) or (j == N - 2)) and i != 2 and i != N-1:
        line += 'I'
      else:
        line += ' '
    print(line)

generateBorderBox(9)
'''
IIIIIIIII
I       I
I IIIII I
I I   I I
I I * I I
I I   I I
I IIIII I
I       I
IIIIIIIII
'''
generateBorderBox(13)
'''
IIIIIIIIIIIII
I           I
I IIIIIIIII I
I I       I I
I I       I I
I I       I I
I I   *   I I
I I       I I
I I       I I
I I       I I
I IIIIIIIII I
I           I
IIIIIIIIIIIII
'''